lib/modulr/api/payments_service.rb

```diff
@@ -6,7 +6,7 @@ class PaymentsService < Service
       def find(id:)
         response = client.get("/payments", { id: id })
         payment_attributes = response.body[:content]&.first
-        raise NotFoundError, "Payment #{id} not found" unless payment_attributes
+        raise ClientError, "Payment #{id} not found" unless payment_attributes
```

Comment:
Should all our `ClientErrors` include a response and a status code too? 🤔

Reply:
My question is if we should raise this error here... Shouldn't we receive a 404 error here?